packages/cli/src/commands/types/generate/actions.test.ts

```diff
@@ -815,3 +818,138 @@ describe('generateStoryblokTypes', () => {
+    expect(files[0].content).toContain('// DO NOT MODIFY THIS FILE BY HAND.');
+    expect(files[0].name).toBe('Hero Section X')
+    expect(files.some(f => f.content.includes('export interface HeroSectionX'))).toBe(true);
+    expect(files[1].name).toBe('cta-button')
+    expect(files.some(f => f.content.includes('export interface CtaButton'))).toBe(true);
+  });
+
+  it('scopes Storyblok imports per file (only files that need Multilink import should have it)', async () => {
+    const withMultilink: SpaceComponent = {
+      name: 'link-card',
+      created_at: '2023-01-01T00:00:00Z',
+      updated_at: '2023-01-01T00:00:00Z',
+      id: 1,
+      schema: {
+        link: { type: 'multilink', required: false, email_link_type: true, asset_link_type: true },
+      },
+    };
+
+    const simpleText: SpaceComponent = {
+      name: 'plain-text',
+      created_at: '2023-01-01T00:00:00Z',
+      updated_at: '2023-01-01T00:00:00Z',
+      id: 2,
+      schema: {
+        text: { type: 'text', required: false },
+      },
+    };
+
+    const spaceData: SpaceComponentsData = {
+      components: [withMultilink, simpleText],
+      datasources: [],
+      groups: [],
+      presets: [],
+      internalTags: [],
+    };
+
+    const res = await generateTypes(spaceData, { separateFiles: true, strict: false });
+    const files = res as { name: string; content: string }[];
+
+    const fileWith = files.find(f => f.name.includes('link-card') || f.content.includes('export interface LinkCard'));
+    const fileWithout = files.find(f => f.name.includes('plain-text') || f.content.includes('export interface PlainText'));
+
+    expect(fileWith).toBeDefined();
+    expect(fileWithout).toBeDefined();
+
+    // Multilink import only in the file that needs it
+    expect(fileWith!.content).toMatch(/import type \{ StoryblokMultilink \} from '\.\.\/storyblok\.d\.ts';/);
+    expect(fileWithout!.content).not.toMatch(/import type \{ StoryblokMultilink \} from '\.\.\/storyblok\.d\.ts';/);
+  });
+
+  it('saveTypesToFile writes one file per component when separateFiles = true and ignores filename', async () => {
+    vi.clearAllMocks();
+
+    const files = [
+      { name: 'Hero', content: '// hero' },
+      { name: 'Button', content: '// button' },
+    ];
+
+    await saveTypesToFile('12345', files, { separateFiles: true, filename: 'IGNORED' });
+
+    // Two writes
+    expect(saveToFile).toHaveBeenCalledTimes(2);
+
+    // join() receives the sanitized filenames (we only mocked return; assert args)
+    expect(join).toHaveBeenCalledWith(expect.any(String), 'Hero.d.ts');
+    expect(join).toHaveBeenCalledWith(expect.any(String), 'Button.d.ts');
+
+    // Should not try to use the single-file filename
+    expect(join).not.toHaveBeenCalledWith(expect.any(String), 'IGNORED.d.ts');
+  });
+
+  it('default behavior still returns a single string when separateFiles is not set', async () => {
```

Comment:
nitpick: I think this test is unnecessary because we already test that the default behavior works with preexisting tests.